### PR TITLE
Add doc about secrets and templates for applyConfig

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @openshift/openshift-team-developer-productivity-test-platform 

Follow up of https://github.com/openshift/release/pull/5384/files#r334517991

### CODE REVIEW

File: clusters/README.md [hunk 1]
```diff
@@ -4,3 +4,40 @@ This folder includes the resources for installation and configuration of [CI-clu
...
+## Best practice for applyConfig
+
+* All the openshift/k8s assets are static and checked into the repo except sensitive information, e.g., github token. [`applyConfig`](https://github.com/openshift/ci-tools/tree/master/cmd/applyconfig) will resolve secrets by auto-populating environment variables for templates.
+
+    we use [bitwarden](https://bitwarden.com/) to store sensitive information. Running bash script _locally_, e.g., [populate-secrets-from-bitwarden.sh](../ci-operator/populate-secrets-from-bitwarden.sh) will create secrets in the cluster the current `oc-cli` logs into. All openshift/k8s assets can then use secrets for referring the sensitive information.
+
+
+* a pre-processing step will be automated and the output of that will be static files that can be checked in. For the case of referring a piece of sensitive information, such as,
+
+    ```
```
Comment (stevekuznetsov) [inline clusters/README.md]: nit: use yaml highlighting on these

Comment (hongkailiu) [reply clusters/README.md]: Done.

File: clusters/README.md [hunk 2]
```diff
@@ -4,3 +4,40 @@ This folder includes the resources for installation and configuration of [CI-clu
 
 * [build-clusters](./build-clusters)
     * [01_cluster](./build-clusters/01_cluster): build cluster managed by DPTP-team.
+
+## Best practice for applyConfig
+
+* All the openshift/k8s assets are static and checked into the repo except sensitive information, e.g., github token. [`applyConfig`](https://github.com/openshift/ci-tools/tree/master/cmd/applyconfig) will resolve secrets by auto-populating environment variables for templates.
+
+    we use [bitwarden](https://bitwarden.com/) to store sensitive information. Running bash script _locally_, e.g., [populate-secrets-from-bitwarden.sh](../ci-operator/populate-secrets-from-bitwarden.sh) will create secrets in the cluster the current `oc-cli` logs into. All openshift/k8s assets can then use secrets for referring the sensitive information.
```
Comment (petr-muller) [inline clusters/README.md]:
```suggestion
    We use [bitwarden](https://bitwarden.com/) to store sensitive information. Running bash script _locally_, e.g., [populate-secrets-from-bitwarden.sh](../ci-operator/populate-secrets-from-bitwarden.sh) will create secrets in the cluster the current `oc-cli` logs into. All openshift/k8s assets can then use secrets for referring the sensitive information.
```

File: clusters/README.md [hunk 3]
```diff
@@ -4,3 +4,40 @@ This folder includes the resources for installation and configuration of [CI-clu
...
+
+
+* a pre-processing step will be automated and the output of that will be static files that can be checked in. For the case of referring a piece of sensitive information, such as,
```
Comment (petr-muller) [inline clusters/README.md]:
```suggestion
* A pre-processing step will be automated and the output of that will be static files that can be checked in. For the case of referring a piece of sensitive information, such as,
```

Comment (hongkailiu) [reply clusters/README.md]: Done.

**Comment by hongkailiu** [comment]
Rebased ... no conflicting though.

**Pull request merged**